funcfile.go

```diff
@@ -72,8 +72,10 @@ type funcfile struct {
 
 func (ff *funcfile) ImageName() string {
 	fname := ff.Name
-	if !strings.Contains(fname, "/") {
-		// then we'll prefix FN_REGISTRY
+
+	// prefix FN_REGISTRY, if provided (important) AND if the image does not
+	// itself contain a registry.
+	if len(strings.SplitN(fname, "/", 3)) < 3 {
```

Comment:
`localhost:5000/mygroup/myimage` would match this though?

Reply:
this one works https://play.golang.org/p/eY2LSGhSDr7 but i think ultimately this logic is flawed trying to have FN_REGISTRY either be a username, a registry, or a registry with username. it seems like the proper fix is making FN_REGISTRY _be_ a registry, and having an additional FN_REGISTRY_USERNAME for cli convenience shenanigans, maybe something slightly easier there but I'm not sure.

Reply:
yeah I'm all for moving to more explicit values for those vars. :+1: